packages/reakit/src/Hidden/HiddenContainer.tsx

```diff
@@ -1,23 +1,30 @@
-import React from "react";
-import PropTypes from "prop-types";
-import { Container } from "constate";
+import * as React from "react";
+import * as PropTypes from "prop-types";
+import { Container, ContainerProps } from "constate";
 
-export const initialState = { visible: false };
+type ContainerState = {
+  visible: boolean;
+};
+
+export const initialState: ContainerState = { visible: false };
 
-export const toggle = () => state => ({ visible: !state.visible });
+export const toggle = () => (state: ContainerState) => ({
+  visible: !state.visible
+});
 
 export const show = () => () => ({ visible: true });
 
 export const hide = () => () => ({ visible: false });
 
-const HiddenContainer = props => (
+const HiddenContainer = (props: ContainerProps<ContainerState>) => (
```

Comment:
Can we try to make it as stated in https://github.com/diegohaz/constate#typescript

Here's an example: https://github.com/reakit/reakit/blob/62b108cd9f9656512ac1bda3d166fa33196de955/packages/website/src/containers/ThemeContainer.tsx#L1-L31

Reply:
Nice 👍 
I will use this, thanks for note